aten/src/ATen/cuda/CUDAGraph.cu

```diff
@@ -0,0 +1,38 @@
+#include <ATen/cuda/CUDAGraph.h>
+
+namespace at::cuda {
+
+__global__ void dynamic_graph_updater_kernel(cudaGraphKernelNodeUpdate* updates, size_t num_updates) {
+    constexpr size_t buf_sz = 1024;
+    __shared__ const void* pointer_indirection[buf_sz];
+    for (size_t off = 0; off < num_updates; off += buf_sz) {
+        int local_num_updates = min(buf_sz, num_updates - off);
+        for (int i = 0; i < local_num_updates; i++) {
+            cudaGraphKernelNodeUpdate& this_update = updates[i + off];
+            if (this_update.field == cudaGraphKernelNodeFieldParam) {
+                if (this_update.updateData.param.size != sizeof(void*)) {
+                    printf("dynamic graph updater only supports pointers\n");
+                    __trap();
```

Comment:
You probably want CUDA_KERNEL_ASSERT_MSG() here and in the other __trap() usage.